### PR TITLE
Support for documenting React components.

### PR DESCRIPTION
**Pull request by gbbr**
We are using `dgeni-alive` in our open-source project, [Superdesk](https://www.superdesk.org/), and it's been working great so far. So thank you for writing this great pre-configuration to `dgeni`!

We had some things that we needed to modify in order for it to suit our project better, so we've done that in our fork (on my account). I was thinking that these changes could benefit others too, who are in the same situation. By Google-ing around, I've noticed that some people are.

In this PR:

* Support for reading JSX files (no transforming).
* Support for new `@ngdoc` type called 'React' to allow documenting React components used in Angular projects

What do you think?

### CODE REVIEW

**Comment by wingedfox** [comment]
Hi,

Thank you for the PR.

I've updated it a bit by moving react-related components to jsx package.
You have to add jsx as an optional package in your documentation building task, but now you're free to implement your own `extractJSDocCommentsProcessor`, using `estraverse-fb` tool to extract comments.

**Comment by gbbr** [comment]
Yes, what you did looks great! Thanks!

**Comment by wingedfox** [comment]
Thanks.

Could you, please, add PR with fix for `extractJSDocCommentsProcessor` and update docs accordingly?

**Comment by gbbr** [comment]
I am quite caught up with work, but I will try to submit a PR when I get a chance. Not sure exactly what it is that you are asking for. These changes alone have helped me solve my problem and documentation is correctly extracted from JSX files in our app now...

Basically this was all I needed, besides including `estraverse-fb` in our project. Not very familiar with dgeni but if you can be more specific I could try and help (can't promise anything though! 😄 )

**Comment by wingedfox** [comment]
As long as you have the change that fixes your issue with broken `eslint`, I'd be glad to see it included in the project.
If all you need is to have `estraverse-fb` together with `es-lint`, then it is an optional dependency and you have to check for its' existence in and display a corresponding warning.

**Comment by gbbr** [comment]
@wingedfox the change is already included. I have actually ended up spending hours to try and understand why having eslint as a peer dependency causes the issue, and I have not figured it out. After adding `estraverse-fb` to our project, it worked out. But the odd thing is that `estraverse-fb` was not needed when `eslint` was not there. So if we were to add a warning to dgeni-alive about this, it might be unneeded in some scenarios (afaik). This probably needs more thorough testing. 

My time is limited but I'm glad to help if you have any ideas about directions we should go towards.

To summarise: dgeni-alive had no problem parsing JSX files in its current state, even without estraverse-fb using eslint 0.9.x, but not using the latest eslint.
